src/debug/main.c

```diff
@@ -44,8 +67,8 @@ int	minishell_loop(t_data *data)
 	lexer_to_simple_cmds(&data->lexer, &data->simple_cmds);
 	expander(data);
 	heredoc(data);
-	exec(data, data->simple_cmds);
-	// print_simple_cmds(&data->simple_cmds);
+	if (g_signal == 0)
+		exec(data, data->simple_cmds);
```

Comment:
Execution is launched only if g_signal is 0 as other value (SIGUSR1) would indicate that the heredoc was interrupted with SIGINT (Ctrl + C) and the program should jump to a new prompt.